### PR TITLE
Feed animal event

### PR DESCRIPTION
**Pull request by hfreeb**
**SpongeAPI** | [SpongeCommon](https://github.com/SpongePowered/SpongeCommon/pull/1255)

This is a response to the feature request #1382, adding an event that is fired when a player feeds an animal.

### CODE REVIEW

File: src/main/java/org/spongepowered/api/event/entity/FeedAnimalEvent.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+     * Fired when an {@link Animal} is fed with food which heals it.
+     */
+    interface Healing extends FeedAnimalEvent {}
+
+    /**
+     * Fired when an child {@link Animal} is fed with a breeding item leading to it aging.
+     */
+    interface Aging extends FeedAnimalEvent {}
+
+}
```
Comment (kashike) [inline src/main/java/org/spongepowered/api/event/entity/FeedAnimalEvent.java]: Missing newline.

File: src/main/java/org/spongepowered/api/event/entity/FeedAnimalEvent.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.item.ItemType;
+import org.spongepowered.api.util.annotation.eventgen.GenerateFactoryMethod;
+
+/**
+ * Fired when a {@link Player} feeds an {@link Animal}.
+ */
+@GenerateFactoryMethod
+public interface FeedAnimalEvent extends InteractEntityEvent {
+    /**
```
Comment (kashike) [inline src/main/java/org/spongepowered/api/event/entity/FeedAnimalEvent.java]: Missing newline.

**Review by kashike — CHANGES_REQUESTED** [review]

File: src/main/java/org/spongepowered/api/event/entity/FeedAnimalEvent.java (hunk 1 above)
Comment (hfreeb) [reply src/main/java/org/spongepowered/api/event/entity/FeedAnimalEvent.java]: Fixed

File: src/main/java/org/spongepowered/api/event/entity/FeedAnimalEvent.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ */
+@GenerateFactoryMethod
+public interface FeedAnimalEvent extends InteractEntityEvent {
+
+    /**
+     * Gets the {@link ItemType} fed to the {@link Animal}.
+     *
+     * @return The item fed to the animal
+     */
+    ItemType getItem();
```
Comment (liach) [inline src/main/java/org/spongepowered/api/event/entity/FeedAnimalEvent.java]: You should provide an `ItemStackSnapshot`.

**Comment by gabizou** [comment]
Might resuscitate this if you're up for absorbing the changes @Zidane for your breeding event changes.

**Comment by JBYoshi** [comment]
Note: This PR was automatically closed because the `bleeding` branch was deleted. It will need to be reopened to target `api-8` instead.  See https://github.com/SpongePowered/SpongeAPI/pull/1428#issuecomment-539689855.

**Pull request closed without merging**